core/services/relay/evm/types/types.go

```diff
@@ -20,11 +20,45 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/utils"
 )
 
+type RelayOpts struct {
+	// TODO BCF-2508 -- should anyone ever get the raw config bytes that are embedded in args? if not,
+	// make this private and wrap the arg fields with funcs on RelayOpts
+	relaytypes.RelayArgs
+	c *RelayConfig
+}
+
+type ChainReaderConfig struct {
+	// ChainContractReaders key is contract name
+	ChainContractReaders map[string]ChainContractReader `json:"chainContractReaders"`
+}
+
+type ChainContractReader struct {
+	ContractABI string `json:"contractABI"`
+	// ChainReaderDefinitions key is chainAgnostic read name.
+	ChainReaderDefinitions map[string]ChainReaderDefinition `json:"chainReaderDefinitions"`
+}
+
+type ChainReaderDefinition struct {
+	ChainSpecificName string         `json:"chainSpecificName"` // chain specific contract method name or event type.
+	Params            map[string]any `json:"params"`
+	ReturnValues      []string       `json:"returnValues"`
+	CacheEnabled      bool           `json:"cacheEnabled"`
+	ReadType          `json:"readType"`
```

Comment:
Is the embedding doing anything for us here? I'm worried it will cause problems later. For example if a `String()` method were added, then `ChainReaderDefinition` would inherit it, and something like log lines would change implicitly.
```suggestion
	ReadType	ReadType          `json:"readType"`
```

Reply:
Yeah, no reason for embedding